packages/dbgpt-core/src/dbgpt/core/interface/tests/test_variables.py

```
@@ -65,6 +65,28 @@ def test_storage_variables_provider():
     assert loaded_variable_value == value
 
 
+def test_storage_variables_provider_bool_value():
+    storage = InMemoryStorage()
+    encryption = SimpleEncryption()
+    provider = StorageVariablesProvider(storage, encryption)
+
+    for i, value in enumerate([True, False]):
+        full_key = f"${{key{i}:name@global}}"
+        id = VariablesIdentifier.from_str_identifier(full_key)
```

Ruff (0.15.21)

[error] Variable `id` is shadowing a Python builtin

(A001)